docs/how-to-guides/how-to-test-contract-with-clarinet-sdk.md

```diff
@@ -0,0 +1,516 @@
+    (ok new-val)
+  )
+)
+
+(define-read-only (read-count)
+  (ok (var-get count))
+)
+```
+
+### Migrating between Clarinet 1 and Clarinet 2
+
+> Note: Clarinet 2 will be released in October 2023, and will create the right boilerplate files. But if a project has been created with Clarinet 1, this prepare to project to run the SDK and Vitest.
+
+Executing this script in a Clarinet 1 project will initialise NPM and Vitest. It will also create a sample test file.
+
+```console
+npx @hirosystems/clarinet-sdk@latest
+```
+
+This script wil lask you if you want to run npm install now, you can press enter to do so.
+It can take a few seconds.
+
+The file `tests/counter_test.ts` that was created by `clarinet contract new counter` can be deleted.  
+
+You can have a look at `tests/contract.test.ts`, it's a sample file showing how to use the SDK with Vitest.
+It can safely be deleted.
+
+
+### Unit tests for `counter` example
+
+Create a file `tests/counter.test.ts` with the following content:
+
+```ts
+import { Cl } from "@stacks/transactions";
+import { describe, expect, it } from "vitest";
+
+const accounts = vm.getAccounts();
+const address1 = accounts.get("wallet_1")!;
+
+describe("test increment method", () => {
+  it("increments the count by the given value", () => {
+    const res1 = vm.callPublicFn("counter", "increment", [Cl.uint(1)], address1);
+
+    console.log(Cl.prettyPrint(res1.result)) // (ok u2)
+
+    expect(res1.result).toBeOk(Cl.uint(2));
+
+    const count1 = vm.getDataVar("counter", "count");
+    expect(count1).toBeUint(2);
+
+    vm.callPublicFn("counter", "increment", [Cl.uint(40)], address1);
+    const count2 = vm.getDataVar("counter", "count");
+    expect(count2).toBeUint(42);
+  });
+
+  it("sends a print event", () => {
+    const res = vm.callPublicFn("counter", "increment", [Cl.uint(1)], address1);
+
+    expect(res.events).toHaveLength(1);
+    const printEvent = res.events[0];
+    expect(printEvent.event).toBe("print_event");
+    expect(printEvent.data.value).toBeTuple({
+      object: Cl.stringAscii("count"),
+      action: Cl.stringAscii("incremented"),
+      value: Cl.uint(2),
+    });
+  });
+});
+```
+
+There is a very important thing happening under the hood. The `vm` object is available globally in the tests, and is automatically initialized before each test.
+
+> You don't need to know much more about that, but if you want to know in details how it works, you can have a look at the `vitest.config.js` file at the root of you project.
+
+We just implement two tests:
+- The first one checks that the `increment` function returns the new value and saves it to the `count` variable.
+- The second one checks that an `print_event` is emitted when the increment function is called.
+
+> You can use `Cl.prettyPrint(value: ClarityValue)` to format any Clarity value into readable Clarity code. It can be useful to debug functions results or events values.
```

Comment:
Great feature, this prettyPrint!

Also, nitpick: `Cl` is an ambiguous namespace. Is it short for Clarinet? Is it a bad idea to reserve just `C` or `clar`?

Reply:
@mefrem `Cl` is directly imported from [Stacks.js](https://stacks.js.org/modules/_stacks_transactions.Cl)